src/dwn-server.ts

```diff
@@ -23,17 +23,6 @@ export class DwnServer {
   }
 
   async listen(): Promise<void> {
-    if (!this.dwn) {
```

Comment:
not sure if we want to add some check to make sure `this.dwn` gets defined or make it non-optional. It's always initialized in main now (or an error is thrown before `listen()` is called)

Reply:
that's a valid concern, as previously there was a check here in `listen` before assigning to `this.dwn`. Since you basically extended and moved that part of logic, I think you need to include that check as well

Reply:
refactored to make dwn non-optional.